forc-pkg/src/pkg.rs

```diff
@@ -379,6 +386,31 @@ fn find_proj_node(graph: &Graph, proj_name: &str) -> Result<NodeIx> {
     }
 }
 
+/// Check minimum forc version given in the manifest file and the current toolchain version
+///
+/// If required minimum forc version is higher than current forc version return an error with
+/// upgrade instructions
+fn validate_version(proj_manifest: &ManifestFile) -> Result<()> {
+    match &proj_manifest.project.forc_version {
+        Some(min_forc_version) => {
+            // Get the current version of the toolchain
+            let crate_version = env!("CARGO_PKG_VERSION");
```

Comment:
This only works when you're building via `cargo` -- when you install a binary and use it elsewhere, this env var will no longer be available unfortunately.

In the past I have had to get the version at compile time. What I did then was I parsed the `Cargo.toml` file using `toml-rs` inside of `build.rs` and output just the version as a text file. In my code, I then did `include_str!()` to get that file, which was just a string representation of the version, at compile time. I think it's the cleanest solution. I'm happy to help you write it out, as I've done it before.

Reply:
> This only works when you're building via cargo -- when you install a binary and use it elsewhere, this env var will no longer be available unfortunately.

Oh, that is a bummer, thanks for the review. I will try to get it like your suggestion and can ping you if I get stuck in the process.

Reply:
I think we probably do want to use this original approach.

@sezna keep in mind that [the `env!` macro](https://doc.rust-lang.org/std/macro.env.html) is evaluated at *compile time*, so it won't matter whether or not the env var exists on the system after installation etc.

Using this env var is also the same approach that `clap` takes with [its `crate_version!` macro](https://docs.rs/clap/latest/clap/macro.crate_version.html).

Reply:
Ohhh. I thought that while `env!` is expanded at compile time, all it is doing is getting the runtime environment variable. I stand corrected, and I'm sorry for the churn that I have caused. 😓